src/elements/select/utils.ts

```diff
@@ -24,8 +24,20 @@ export const applySearchHighlight = (options: string[], value: string) => {
   const matches = [];
   const noMatches = [];
 
+  // make sure the value is fixed for special characters
+  let valueCopy = value;
+  let i = 0;
+  while (i < valueCopy.length) {
+    if (/[^\dA-Za-z]/.test(valueCopy[i] as string)) {
+      /* eslint-disable no-useless-escape */
+      valueCopy = `${valueCopy.slice(0, i)}\\${valueCopy.slice(i, valueCopy.length)}`;
+      i += 1;
+    }
+    i += 1;
+  }
```

Comment:
[opt] this feels hard to read, would add a quick comment explaining the logic more specifically here (prepend special characters with '/' to create appropriate regex). any reason you didn't construct `valueCopy` while iterating over `value`? might simplify this a bit

Reply:
yeah I can change it to do that too! just did the quickest that came to mind but will add thanks!